File: 프로그래머스/2/87946. 피로도/피로도.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;
int answer = -1;
int solution(int k, vector<vector<int>> d) {
    sort(d.begin(),d.end());
    
    do{
        int stress = k;
        int count = 0;
        for(int i=0;i<d.size();i++){
            if(stress >= d[i][0]){
                stress -= d[i][1];
                count ++;
            }
        }
        
        answer = max(answer, count);
    }while(next_permutation(d.begin(),d.end()));
    
    return answer;
}

File: 프로그래머스/2/87946. 피로도/피로도.py
# Python 풀이 : Crush on Study
from itertools import permutations
def solution(k, dungeons):
    answer = -1
    save_k = k
    # k는 현재 피로도
    result = list(permutations(dungeons,len(dungeons)))    
    for idx in result:
        cnt = 0
        for need,use in idx:
            if save_k>=need: # 최소 피로도조건보다 같거나 큰지?
                save_k-=use
                cnt += 1
                answer = max(answer,cnt)

            else:
                save_k = k
                break

    return answer
